docs/source/tutorial_optimization.md

```diff
@@ -0,0 +1,849 @@
+```
+
+Comparing the result to the FOM model, we see that the number of
+iterations and evaluations of the model slightly decreased. As expected,
+we see that the optmization routine is very fast because the surrogate
+enables almost instant evaluations of the primal equation.
+
+As mentioned above, we should not forget that we required the offline
+time to build our surrogate. In our case, the offline time is still low
+enough to get a speed up over the FOM optimization. Luckily,
+`atol=1e-2` was enough to achieve an absolute error of roughly `1e-06`
+but it is important to notice that we do not know this error before
+choosing `atol`.
+
+To show that the ROM optimization roughly followed the same path as the
+FOM optimization, we visualize both of them in the following plot.
+
+```{code-cell}
+reference_plot = plot_3d_surface(fom_objective_functional, XX, YY, alpha=0.5)
+reference_plot_mean_z_lim = 0.5*(reference_plot.get_zlim()[0] + reference_plot.get_zlim()[1])
+
+for mu in reference_minimization_data['evaluation_points']:
+    addplot_xy_point_as_bar(reference_plot, mu[0], mu[1], color='green',
+                            z_range=(reference_plot.get_zlim()[0], reference_plot_mean_z_lim))
+
+for mu in RB_minimization_data['evaluation_points']:
+    addplot_xy_point_as_bar(reference_plot, mu[0], mu[1], color='orange',
+                           z_range=(reference_plot_mean_z_lim, reference_plot.get_zlim()[1]))
+```
+
+## Computing the gradient of the objective functional
+
+A major issue of using finite differences for computing the gradient of
+the objective functional is the number of evaluations of the objective
+functional. In the FOM example from above we saw that
+many evaluations of the model were only due to the
+computation of the finite differences. If the problem is more complex
+and the mesh is finer, this can lead to a serious waste of
+computational time. Also from an optimizational point of view it is
+always better to compute the true gradient of the objective functional.
+
+For computing the gradient of the linear objective functional
+{math}`\mathcal{J}(\mu)`, we can write for every direction
+{math}`i= 1, \dots, P`
+
+```{math}
+\begin{align} \label{gradient:sens} \tag{1}
+d_{\mu_i} \mathcal{J}(\mu) = \partial_{\mu_i} J(u_{\mu}, \mu) + \partial_u J(u_{\mu}, \mu)[d_{\mu_i} u_{\mu}]
+   =   \partial_{\mu_i} J(u_{\mu}, \mu) + J(d_{\mu_i} u_{\mu}, \mu)
+\end{align}
+```
+
+Thus, we need to compute the derivative of the
+solution {math}`u_{\mu}` (also called sensitivity). For this, we need to
+solve another equation: Find {math}`d_{\mu_i} u_{\mu} \in V`, such that
+
+```{math}
+ \label{sens} \tag{2}
+a_\mu(d_{\mu_i} u_{\mu}, v) = \partial_{\mu_i} r_\mu^{\text{pr}}(u_{\mu})[v] \qquad \qquad \forall v \in V
+```
+
+where {math}`r_\mu^{\text{pr}}` denotes the residual of the primal
+equation, i.e.
+
+```{math}
+r_\mu^{\text{pr}}(u)[v] := l_\mu(v) - a_\mu(u, v) &&\text{for all }v \in V
```

Comment:
```suggestion
r_\mu^{\text{pr}}(u)[v] := l_\mu(v) - a_\mu(u, v) \qquad \qquad \text{for all }v \in V
```
It seems that the `&&` causes some problems such that this gets not displayed correctly.

Reply:
Problem already in 2020.2.0 btw, so just like https://github.com/pymor/pymor/pull/1323#issuecomment-848604803, not introduced with switch to markdown.